JavaScript/JavaScriptSDK/appInsights.ts

```diff
@@ -40,54 +40,58 @@ module Microsoft.ApplicationInsights {
         public static defaultConfig: IConfig;
 
         constructor(config: IConfig) {
-            this.config = config || <IConfig>{};
-
-            // load default values if specified
-            var defaults: IConfig = AppInsights.defaultConfig;
-            if (defaults !== undefined) {
-                for (var field in defaults) {
-                    // for each unspecified field, set the default value
-                    if (this.config[field] === undefined) {
-                        this.config[field] = defaults[field];
+            try {
```

Comment:
We don't want it in the constructor - let it fail, it'll be unexpected to see object constructed with undefined state.